src/features/defi/providers/fox-farming/hooks/useFoxFarming.ts

```diff
@@ -322,14 +326,18 @@ export const useFoxFarming = (
       to: contractAddress,
       value: '0',
       wallet,
+      checkLedgerAppOpenIfLedgerConnected,
     })
 
-    await checkLedgerAppOpenIfLedgerConnected(ethChainId)
-
     const txid = await buildAndBroadcast({
       adapter,
       buildCustomTxInput,
-      receiverAddress: await adapter.getAddress({ accountNumber, wallet }),
+      receiverAddress: await adapter.getAddress({
+        accountNumber,
+        wallet,
+        checkLedgerAppOpenIfLedgerConnected,
+      }),
+      checkLedgerAppOpenIfLedgerConnected,
     })
```

Comment:
This is pretty wild. I'm starting to wonder if `EvmBaseAdapter` could have `checkLedgerAppOpenIfLedgerConnected` as a class method?

Reply:
That would be ideal indeed! Unfortunately, the whole reason why we pass this down is because `checkLedgerAppOpenIfLedgerConnected` is a react-land callback, coming from a hook.

Whle the core wallet logic could absolutely live there, not really sure how we would solve opening/closing the modal with such indirection if this was implemented in chain-adapters directly 🤔